### PR TITLE
Major build system refactor for spec.yaml.io

### PR DESCRIPTION
**Pull request by ingydotnet**

### CODE REVIEW

File: tool/bin/markydown-to-kramdown [hunk 1]
```diff
@@ -371,29 +371,36 @@ sub fmt_table {}
...
+      }
+    }
   }
-  else {
-    $front = "---\n---\n";
+
+  if (not defined $front) {
+    my $layout = 'default';
+    if ($markdown =~ /\A# YAML Ain.t Markup Language/) {
+      $layout = 'spec';
```
Comment (Thom1729) [inline tool/bin/markydown-to-kramdown:388]: What's the purpose of this?

File: www/build/_layouts/default.html [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+<section>
+
+{{content}}
+
+</section>
+</div>
+</main>
+
+<script>
+for (const c of document.querySelectorAll('code > code')) {
```
Comment (Thom1729) [inline www/build/_layouts/default.html:24]: What's the purpose of this?

File: www/build/favicon.svg [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg
+	xmlns="http://www.w3.org/2000/svg"
+	xmlns:cc="http://creativecommons.org/ns#"
```
Comment (Thom1729) [inline www/build/favicon.svg:4]: Do we need any of these xmlns:foo attributes?

**Review by Thom1729 — COMMENTED** [review]
The make stuff is mostly impenetrable to me without a lot more reading, but I've perused the rest.

**Comment by Thom1729** [comment]
When running `make -C www clean serve` on this branch, `localhost/` redirects to `localhost/main`, which 404s.

File: tool/bin/markydown-to-kramdown (hunk 1 above)
Comment (ingydotnet) [reply tool/bin/markydown-to-kramdown:388]: Will add comment to fix.

File: www/build/_layouts/default.html (hunk 2 above)
Comment (ingydotnet) [reply www/build/_layouts/default.html:24]: Will add comment to fix.

File: www/build/favicon.svg (hunk 3 above)
Comment (ingydotnet) [reply www/build/favicon.svg:4]: fixed

**Review by Thom1729 — APPROVED** [review]

**Pull request merged**